### DISCUSSION TITLE
Allow dash to appear in hashtag

### DISCUSSION THREAD

**Discussion by jotaen** [jotaen/klog#108] [Feature ideas & discussions]
Currently, only letters, digits, and underscores are allowed in hashtags.

I’m personally a big fan of [kebab case](https://en.wiktionary.org/wiki/kebab_case), so I wonder whether there could be any disadvantage in extending the spec to allow dashes to appear in hashtags. So you could write all of the following:

- `#helloWorld`
- `#hello_world`
- `#hello-world`

I still don’t want to allow other punctuation such as dots or colons, because it’s important that hashtags can be used within sentences, e.g.:

```
2020-01-01
  1h Went to the #gym!
  1h Called #andy, we discussed politics
  1h #chores: dishes, laundry, windows
```

**Comment by vladdeSV** [comment]
I would like to add that this will be a breaking change, as I have some tags which are directly followed by a dash.

In Swedish, when combining words from different languages or shortenings, they are joined by a dash.

For instance, "ActiveDirectory integration" would be written (including shortening) in Swedish as "AD-integrationen". Because dashes are not part of tags, I have written sentences which are similar to (example):

    1h fixat med #ad-integrationen

This would now result in a  new tag `#ad-integrationen`.

Personally I am a big fan of kebab-case, and would like to see this in the next spec release. Just letting you know of possible incompatibilities.

**Comment by jotaen** [comment]
Yeah, I’m also not sure yet whether this works out. I think punctuation rules are quite different between languages, so I probably need to research this a bit to see how “safe” this proposal is.

In english, for example, the [em-dash](https://en.wikipedia.org/wiki/Dash#Em_dash) is also used without surrounding spaces. So if people used a regular dash instead of an em-dash for that, it would also break their tags.

Twitter, btw., doesn’t allow any special characters to appear in tags, only letters and digits.

**Comment by jotaen** [comment]
(Actually this is a feature discussion, so moving this over to Discussions.)

**Comment by jotaen** [comment]
For reference: resolved via https://github.com/jotaen/klog/pull/171
